### PR TITLE
Stop dropping metadata fields in span documents

### LINKED ISSUE #10716: Stop dropping metadata fields in span documents
From https://github.com/elastic/apm-server/pull/10697#pullrequestreview-1400922831:

> I wonder if we should drop that `remove` processor altogether. This has been around since forever, when we used to store a transaction document even for non-sampled transactions, and when we didn't have a default ILM delete phase . I feel like we have better options for storage management these days.

We should investigate how much storage these fields actually take, using Rally, and consider dropping that pipeline altogether so spans and transactions have the same metadata fields.

### PR DESCRIPTION
**Pull request by dmathieu**
## Motivation/summary

We drop these fields for spans, but not for transactions. So not dropping them makes both events more aligned.

## Checklist

- [x] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)
- [x] **N/A** Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)
- [x] **N/A** Documentation has been updated

For functional changes, consider:
- Is it observable through the addition of either **logging** or **metrics**?
- Is its use being published in **telemetry** to enable product improvement?
- Have system tests been added to avoid regression?

## How to test these changes

Run the suite of automated changes.

## Related issues

Closes #10716.

### CODE REVIEW

File: apmpackage/apm/data_stream/traces/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -26,20 +26,3 @@ processors:
       field: event.success_count
       value: 0
       ignore_failure: true
-  # end of event.success_count logic
```
Comment (kruskall) [inline apmpackage/apm/data_stream/traces/elasticsearch/ingest_pipeline/default.yml:29]: **question**: why was this removed ? I think we want to keep this comment

File: changelogs/head.asciidoc [hunk 2]
```diff
@@ -17,3 +17,5 @@ https://github.com/elastic/apm-server/compare/8.9\...main[View commits]
 
 [float]
 ==== Added
+
```
Comment (carsonip) [inline changelogs/head.asciidoc:20]:
```suggestion
```

**Review by carsonip — APPROVED** [review]

**Review by axw — APPROVED** [review]
LGTM!

@AlexanderWert @felixbarny FYI. There's a small storage size increase (see https://github.com/elastic/apm-server/issues/10716), but this is necessary for us to converge transactions and spans.

**Review by felixbarny — APPROVED** [review]

**Comment by axw** [comment]
Verified with 8.10.0-BC1. I sent some transactions and spans and confirmed that the spans had `process.*` fields in tact.
